### PR TITLE
fix: camelyon benchmark

### PR DESCRIPTION
**Pull request by thbcmlowk**
## Related issue

Closes FL-1084

## Summary

* Fixed: `sklearn` → `scikit-learn` in pypi benchmark dependencies
* Fixed: `editable_mode=True` for substra dependencies
* Fixed: pinned pypi dependency versions

```python
    dependencies = Dependency(
        pypi_dependencies=["numpy==1.23.5", "torch==2.0.1", "scikit-learn==1.1.1"],
        local_code=[base / "common", base / "weldon_fedavg.py"],
        # Keeping editable_mode=True to ensure nightly test benchmarks are ran against main substrafl git ref
        editable_mode=True,
    )
```

* Removed: crendentials for remote environment
* Fixed: minor typing issues

## Notes

[Test Camelyon Benchmark](https://github.com/owkin/substra-ci/actions/runs/5533341930)

## Please check if the PR fulfills these requirements

- [x] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: benchmark/camelyon/pure_substrafl/register_assets.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def instantiate_clients(
                 url=organization.get("url"),
                 username=organization.get("username"),
                 password=organization.get("password"),
+                insecure=organization.get("insecure") or False,
```
Comment (guilhem-barthes) [inline benchmark/camelyon/pure_substrafl/register_assets.py:65]: Nitpicking:
```suggestion
                insecure=organization.get("insecure", False),
```

File: benchmark/camelyon/pure_substrafl/register_assets.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def instantiate_clients(
     mode: substra.BackendType = substra.BackendType.LOCAL_SUBPROCESS,
     n_centers: Optional[int] = 2,
     conf: Optional[dict] = None,
```
Comment (guilhem-barthes) [inline benchmark/camelyon/pure_substrafl/register_assets.py:44]: Looks like it's a `Optional[list[dict]]` (same in docstring)

File: benchmark/camelyon/workflows.py [hunk 3]
```diff
@@ -53,7 +54,6 @@ def substrafl_fed_avg(
         n_centers (int): Number of centers to be used for the fed avg strategy.
         learning_rate (int): Learning rate to be used.
         n_rounds (int): Number of rounds for the strategy to be executed.
-        n_local_steps (int): Number of updates for each step of the strategy.
```
Comment (guilhem-barthes) [inline benchmark/camelyon/workflows.py:56]: 👍

File: benchmark/camelyon/pure_substrafl/register_assets.py (hunk 2 above)
Comment (thbcmlowk) [reply benchmark/camelyon/pure_substrafl/register_assets.py:44]: Indeed

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: benchmark/camelyon/workflows.py [hunk 4]
```diff
@@ -110,7 +111,7 @@ def substrafl_fed_avg(
         experiment_folder=Path(__file__).resolve().parent / "benchmark_cl_experiment_folder",
     )
 
-    clients[0].wait_compute_plan(compute_plan.key)
+    clients[0].wait_compute_plan(key=compute_plan.key, raise_on_failure=True)
```
Comment (SdgJlbl) [inline benchmark/camelyon/workflows.py:114]: I'm wondering if while we're at fixing this test, we want to check anything else... 
Execution time has more or less an upper bound due to the timeout allocation, but maybe we should check that whatever performance metrics are returned are between 0 and 1. Maybe we can have a narrower interval given that the seed is fixed.
(Perfectly happy with postponing that to a follow-up issue since the priority now is to get the CI green again)

Comment (thbcmlowk) [reply benchmark/camelyon/workflows.py:114]: Indeed, I was wondering what kind of additional controls could be applied here. This is a great idea. As it will take some time to do it right (ie monitoring metrics results on at least 2 or 3 runs) and there is a urge to fix the CI ; I have created a [follow-up issue](https://linear.app/owkin/issue/FL-1095/check-performance-result-of-camelyon-benchmark).

**Review by SdgJlbl — APPROVED** [review]
Thank you
